compiler/rustc_const_eval/src/util/check_validity_requirement.rs

```diff
@@ -155,6 +155,7 @@ fn check_validity_requirement_lax<'tcx>(
     }
 
     match &this.variants {
+        Variants::Empty => return Ok(false),
```

Comment:
This line is either unreachable or it obviously changes behavior of the code. Not yet sure which.

Reply:
The new logic makes more sense -- empty-variant types are never valid. We just didn't bother specifically checking for this before.

But yeah it may well be unreachable. It doesn't really matter.